R-package/vignettes/the_RGF_package.Rmd

```diff
@@ -0,0 +1,566 @@
+---
+title: "Regularized Greedy Forest in R"
+author: "Lampros Mouselimis"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Regularized Greedy Forest in R}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+
+
+The RGF package is a wrapper of the [*Regularized Greedy Forest*](https://github.com/fukatani/rgf_python) *python* package, which also includes a [*Multi-core implementation (FastRGF)*](https://github.com/baidu/fast_rgf). Portability from Python to R was made possible using the [reticulate](https://github.com/rstudio/reticulate) package and the installation requires basic knowledge of Python. Except for the Linux Operating System, the installation on Macintosh and Windows might be somehow cumbersome (on windows the package currently can be used only from within the command prompt). Detailed installation instructions for all three Operating Systems can be found in the *README.md* file and in the [*rgf_python*](https://github.com/fukatani/rgf_python) Github repository.
+
+<br>
+
+The *Regularized Greedy Forest* algorithm is explained in detail in the paper [*Rie Johnson and Tong Zhang, Learning Nonlinear Functions Using Regularized Greedy Forest*](https://arxiv.org/abs/1109.0887). A small synopsis would be *"... the resulting method, which we refer to as regularized greedy forest (RGF), integrates two ideas: one is to include tree-structured regularization into the learning formulation; and the other is to employ the fully-corrective regularized greedy algorithm ...."*.
+
+<br>
+
+At the time of writing this Vignette (11 - 02 - 2018), there isn't a corresponding implementation of the algorithm in the R language, so I decided to port the Python package in R taking advantage of the reticulate package. In the next lines, I will explain the functionality of the package and I compare RGF with other similar implementations, such as [*ranger*](https://github.com/imbs-hl/ranger) (random forest algorithm) and [*xgboost*](https://github.com/dmlc/xgboost/tree/master/R-package) (gradient boosting algorithm), in terms of time efficiency and error rate improvement.
+
+<br>
+
+#### The RGF package
+
+<br> 
+
+The *RGF* package includes the following R6-classes / functions,
+
+<br>
+
+##### **classes** 
+
+
+<br>
+
+
+|    RGF_Regressor        |   RGF_Classifier       |  FastRGF_Regressor    |  FastRGF_Classifier  |
+| :-----------------:     |  :-----------------:   | :------------------:  | :------------------: |
+|   fit()                 |  fit(()                | fit()                 | fit()                |
```

Comment:
Also, all classes from this table have `set_params()` method. Does R wrapper supports them too?

Reply:
I understand. The RGF R package supports all available methods of the rgf_python wrapper. Not all methods have to be included in this table in order to work. I explained how this can be done in a [past issue](https://github.com/mlampros/RGF/issues/2#issuecomment-397206793).

I included all methods that I thought they might important / necessary to make both RGF and FastRGF work. The *set_params()* method might be tricky because it requires that a dictionary is passed to the method ( dictionary does not exist as an object in R and someone has to use the reticulate::dict() function ).

Can you include a use case, except for the [test cases](https://github.com/RGF-team/rgf/blob/master/python-package/tests/test_rgf_python.py) that I'm aware of, to understand the importance of setting the parameters after the algorithm (RGF, FastRGF) is initialized?

Reply:
Oh, I see. I didn't know that passing a dictionary is rather complicated task. Forget my comment =)